src/bufr/test_ingester.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include <time.h>
+
+#include <iomanip>
+#include <iostream>
+
+#include "eckit/config/YAMLConfiguration.h"
+#include "eckit/filesystem/PathName.h"
+#include "oops/util/IntSetParser.h"
+
+#include "BufrParser/BufrDescription.h"
+#include "BufrParser/BufrMnemonicSet.h"
+#include "BufrParser/BufrParser.h"
+#include "IodaEncoder/IodaDescription.h"
+#include "IodaEncoder/IodaEncoder.h"
+#include "DataContainer.h"
+
+static const char* CONFIG_FILE = "./testinput/bufrtest_bufrdescription.yaml";
+
+
+namespace Ingester
+{
+    void test_createDescriptionManually()
+    {
+        static const char* INPUT_FILE = "./Data/gdas.t12z.1bmhs.tm00.bufr_d";
```

Comment:
Is there a reason this has to be hard-wired?
I understand this is a test program, but still.  If a YAML is associated with this, then can we not get this information from the yaml

Reply:
Yes, this is to test the programatic creation of a BufrDescription object (without using a YAML file). I was careful to write these classes so that you could use them with or without a YAML file. In the future we could (as an example) wrap this library in a python interface and then operate the conversion process completely in python (without creating a YAML file). Not saying we will ever do this, but we do have that flexibility. This test tests that interface.